### ISSUE TITLE
Move Leshan Website to Hugo

### ISSUE THREAD

**Issue by sbernard31** [eclipse-leshan/leshan#1671]
Eclipse foundation stop using php for website : 

> Hi,
> 
>   I wanted to remind everyone that PHP support will be removed at the end of December, and a final PHP brownout on [eclipse.dev](http://eclipse.dev/) is scheduled for December 4th, 2024.
> 
> Thanks to the work that many of you have put in, only 46% of project websites still have some PHP content. 
> 
> We do have a [Hugo based template][1] you can leverage, so please file a [ticket][2] if you have questions, and we'll do what we can t help.  You can also chose to use a different tool to generate your project website if you feel it better fits your needs.
> 
> -Matt.
> 
[1]:https://gitlab.eclipse.org/eclipsefdn/it/webdev/hugo-eclipsefdn-website-boilerplate
[2]:https://gitlab.eclipse.org/eclipsefdn/helpdesk/-/issues

Currently [our web site](https://github.com/eclipse-leshan/leshan-website) seems to use php, so we should migrate to hugo soon. 

Reading [](https://gitlab.eclipse.org/eclipsefdn/it/webdev/hugo-eclipsefdn-website-boilerplate#build-my-projects-website)
There is 2 recommended way by eclipse foundation.
 1. One repo where we mix source and generated website without jenkins job
 2. Two repo, 1 for the code and 1 for the generated website + 1 Jenkins job which will automatically generate static website.
 
 Any opinions ? We don't really edit that website often so maybe **1.** is enough ?

**Comment by jvermillard** [comment]
Due to the nature of website, this could be just plain HTML? Is there really a need for Hugo?

**Comment by sbernard31** [comment]
> Due to the nature of website, this could be just plain HTML? Is there really a need for Hugo?

Yep maybe static is enough.

I didn't double check but I think currently we are using php just for the old official eclipse footer.

Reading https://gitlab.eclipse.org/eclipsefdn/it/webdev/hugo-eclipsefdn-website-boilerplate, I tell myself that maybe reusing the eclipse foundation hugo theme. (which will be maintained by eclipse) will be a good idea. 

Eclipse foundation say : 

> _The preferred static website generator for Eclipse project websites is [Hugo](https://gohugo.io/) and we recommend to our projects that they get started by creating a copy of our [hugo-eclipsefdn-website-boilerplate](https://gitlab.eclipse.org/eclipsefdn/it/webdev/hugo-eclipsefdn-website-boilerplate) project. While you're not obligated to use them, please note that Hugo and hugo-eclipsefdn-website-boilerplate are the supported solutions by the Eclipse Foundation. Using a different technology may result in reduced support. ..._

> _... Although you're starting with [hugo-solstice-theme](https://gitlab.eclipse.org/eclipsefdn/it/webdev/hugo-solstice-theme), remember that Hugo is highly extensible, allowing you to override as much or as little as you need. For example, you may choose to keep our default footer but override our header ..._

**Comment by jvermillard** [comment]
To use Hugo for my personal website, the good: it makes updating things like the footer easy, the bad, it's in 0.x version since forever and upgrades are not always seamless. If it's the eclipse way to go, I suppose they will control the breakages

**Comment by sbernard31** [comment]
I tried to play with Hugo eclipse theme.
Maybe I'm not so good but that was not so easy to tweak it.
So I tried to remove all php code (I'm not even sure that was really used ...) and keep a simple static HTML website + some javascript. (https://github.com/eclipse-leshan/leshan-website/commit/b6177c15f909faadd84576c48c1036d43c4a9265)

Looking at as much our website is simple, I think we could even be able to remove all javascript (e.g. there is still some old jquery usage :scream:) but this is more work.

I tested locally but this is not yet deployed we will see tomorrow if the website still work.
